parser-typechecker/src/Unison/TermParser.hs

```diff
@@ -274,8 +293,54 @@ termLeaf =
     , seq term
     , delayQuote
     , bang
+    , docBlock
     ]
 
+docBlock :: Var v => TermP v
+docBlock = do
+  openTok <- openBlockWith "[:"  
+  segs <- many segment  
+  closeTok <- closeBlock
+  let a = ann openTok <> ann closeTok
+  pure $ Term.app a (Term.constructor a DD.docRef DD.docJoinId) (Term.seq a segs)
+  where
+  segment = blob <|> linky
+  blob = do 
+    s <- string
+    pure $ Term.app (ann s) (Term.constructor (ann s) DD.docRef DD.docBlobId)
+                            (Term.text (ann s) (L.payload s))
+  linky = asum [include, signature, evaluate, source, link]
+  include = do 
+    _ <- P.try (reserved "include")
+    hashQualifiedPrefixTerm
+  signature = do
+    _ <- P.try (reserved "signature")
+    tok <- termLink' 
+    pure $ Term.app (ann tok) 
+                    (Term.constructor (ann tok) DD.docRef DD.docSignatureId) 
+                    (Term.termLink (ann tok) (L.payload tok))
+  evaluate = do
+    _ <- P.try (reserved "evaluate")
+    tok <- termLink' 
+    pure $ Term.app (ann tok) 
+                    (Term.constructor (ann tok) DD.docRef DD.docEvaluateId)
+                    (Term.termLink (ann tok) (L.payload tok))
+  source = do
+    _ <- P.try (reserved "source")
+    l <- link'' 
+    pure $ Term.app (ann l) 
+                    (Term.constructor (ann l) DD.docRef DD.docSourceId)
+                    l 
+  link'' = either ty t <$> link' where
+    t tok = Term.app (ann tok) 
+                     (Term.constructor (ann tok) DD.linkRef DD.linkTermId) 
+                     (Term.termLink (ann tok) (L.payload tok))
+    ty tok = Term.app (ann tok) 
+                      (Term.constructor (ann tok) DD.linkRef DD.linkTypeId) 
+                      (Term.typeLink (ann tok) (L.payload tok))
+  link = d <$> link'' where
+    d tm = Term.app (ann tm) (Term.constructor (ann tm) DD.docRef DD.docLinkId) tm
```

Comment:
Term parser changes. Straightforward.

Reply:
Do you want parser tests for any of these, or are you just catching it in the transcript?

Reply:
Right.